domino_environments/_environments.py

```diff
@@ -325,8 +347,47 @@ def create_revision(
             headers={"Content-Type": "application/x-www-form-urlencoded"},
         )
 
+    #TODO: add some paging logic in here. search 10/20 at a time and fail if not found
+    def get_build_status(self, environment: Environment, revision_id: str = None):
+        """Gather current status of build job for a given environment/revision
+
+        Args:
+            environment: The environment object.
+            revision_id: The ID of the revision, defaults to the environment's active revision.
+        """
+        revision_id = revision_id or environment.active_revision.get("id")
+        summary = self._get_revision_summary(environment.id)
+        return summary['buildStatus'][revision_id]
+
+    def get_build_logs(self, environment: Environment, revision_id: str = None):
+        """Gather logs of build job for a given environment/revision
+
+        Args:
+            environment: The environment object.
+            revision_id: The ID of the revision, defaults to the environment's active revision.
+        """
+        revision_id = revision_id or environment.active_revision.get("id")
+        summary = self._get_revision_summary(environment.id)
+        logs_url = summary['buildLogsUrl'][revision_id]
+
+        url = self._routes.build_logs(logs_url)
+        res = self.request_manager.get(url)
+
+        regex = re.compile((r'<td class="line".*?>(.*)</td>'))
+        logs_matching = regex.findall(res.text)
+
+        return logs_matching
+
+
+
     def _scrape_revision(self, environment_id: str, revision_id: str) -> dict:
         url = self._routes.revision_download(environment_id, revision_id)
         res = self.request_manager.get(url)
         file_io = io.BytesIO(res.content)
         return parse_revision_tar(file_io)
+
+
+    def _get_revision_summary(self, environment_id: str, pageStart=0, pageEnd=500) -> dict:
+        url = self._routes.revision_summaries(environment_id, pageStart, pageEnd)
+        res = self.request_manager.get(url)
+        return res.json()
```

Comment:
Good to have a newline at the end of the file
```suggestion
        return res.json()

```